geoips/geoips_utils.py

```diff
@@ -75,40 +75,30 @@ def load_all_yaml_plugins():
     in ``.yaml``. Read each plugin file
     """
     # Load all entry points for plugin packages
-    plugin_packages = get_entry_point_group("geoips.plugin_packages")
 
-    # Loop over the plugin packages and load all of their yaml plugins
+    reg_plug_path = resources.files("geoips") / "registered_plugins.yaml"
+    if not os.path.exists(reg_plug_path):
+        from . import create_plugin_registry
+
+        create_plugin_registry.main()
+    plugin_packages = get_entry_point_group("geoips.plugin_packages")
     plugins = {}
+    yaml_interfaces = [
+        "feature_annotators",
+        "gridline_annotators",
+        "product_defaults",
+        "products",
+        "sectors",
+    ]
     for pkg in plugin_packages:
-        pkg_plugin_path = resources.files(pkg.value) / "plugins"
-        yaml_files = pkg_plugin_path.rglob("*.yaml")
-
-        # Loop over the yaml files from one package
-        for yaml_file in yaml_files:
-            # Load
-            yaml_plugin = yaml.safe_load(open(yaml_file, "r"))
-
-            # Set some additional information on the YAML plugin
-            # The name of the package the plugin comes from
-            if not yaml_plugin:
-                raise PluginError(
-                    f"YAML file is empty, please fill {yaml_file} with the "
-                    f"appropriate information."
-                )
-            yaml_plugin["package"] = pkg.value
-            # The relative path to the plugin within the package
-            yaml_plugin["relpath"] = str(yaml_file.relative_to(pkg_plugin_path))
-            # Absolute path to the plugin
-            yaml_plugin["abspath"] = str(yaml_file)
```

Comment:
I think in create_plugin registry, we should be finding the relative path like this, not relatie to the current directory.

Reply:
See: https://github.com/NRLMMD-GEOIPS/geoips/pull/380/files
Resolve this conversation if you are good with those changes.

Reply:
I think this comment still needs to be addressed @evrose54 @mindyls

Reply:
After a second read-through this appears to be handled.